src/ClientInstance.cpp

```diff
@@ -384,14 +385,26 @@ PVR_ERROR CVNSIClientInstance::GetChannels(bool radio, kodi::addon::PVRChannelsR
       if (m_protocol >= 6)
       {
         std::string path = CVNSISettings::Get().GetIconPath();
-        std::string ref = vresp->extract_String();
+        std::string ref = vresp->extract_String(); // Picon ref
         if (!path.empty())
         {
           if (path[path.length() - 1] != '/')
             path += '/';
-          path += ref;
-          path += ".png";
-          tag.SetIconPath(path);
+
+          ref += ".png";
+          std::string image_path = path + ref;
+          if (kodi::vfs::FileExists(image_path)) // Use picon
```

Comment:
‘if (!kodi::vfs…`

then you can move the setting of the icon path to below the statement and you don’t need the `else`

Reply:
Thanks for the tip. Better now?

Reply:
Nearly there. See below 😉